state/store.go

```diff
@@ -148,7 +149,7 @@ func LoadABCIResponses(db dbm.DB, height int64) (*ABCIResponses, error) {
 // This is useful in case we crash after app.Commit and before s.Save().
 // Responses are indexed by height so they can also be loaded later to produce Merkle proofs.
 func saveABCIResponses(db dbm.DB, height int64, abciResponses *ABCIResponses) {
-	db.SetSync(calcABCIResponsesKey(height), abciResponses.Bytes())
+	db.Set(calcABCIResponsesKey(height), abciResponses.Bytes())
```

Comment:
we need to flush these to disk incase we crash before we get to call SaveState

Reply:
Though I would have thought the persistence tests would have caught this if it was wrong. Hmph.

Reply:
You're right, we should flush these, cuz otherwise we'll commit to the app and possibly crash, losing the responses.

I think it's difficult to catch this because to catch it, it isn't sufficient to fail after saveABCIResponse(), but we need to also call updateState+blockExec.Commit and then fail (exit) before leveldb finalizes... 

I think we can simulate this by implementing a special wrapper around DB that delays .Set() for many seconds until maybe the next *Sync call.  I'll make an issue of it.

Reply:
https://github.com/tendermint/tendermint/issues/2024